src/Microsoft.SqlTools.ServiceLayer/Scripting/Contracts/ScriptingRequest.cs

```diff
@@ -0,0 +1,63 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.Scripting.Contracts
+{
+    /// <summary>
+    /// Parameters for a script request.
+    /// </summary>
+    public class ScriptingParams
+    {
+        /// <summary>
+        /// Gets or sets the file path used when writing out the script.
+        /// </summary>
+        public string FilePath { get; set; }
+
+        /// <summary>
+        /// Gets or sets connection string of the target database the scripting operation will run against.
+        /// </summary>
+        public string ConnectionString { get; set; }
```

Comment:
I'm going to leave it as is since it's only defined in two request types. If we start seeing it in more request types, I'll make this change.